src/lib/base64.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Converts a string to a base64 string, preserving the UTF-8 encoding
+ */
+export function stringToBase64(str: string) {
+  return bytesToBase64(new TextEncoder().encode(str))
+}
+
+/**
+ * Converts a base64 string to a string, preserving the UTF-8 encoding
+ */
+export function base64ToString(base64: string) {
+  return new TextDecoder().decode(base64ToBytes(base64))
+}
+
+/**
+ * From the MDN Web Docs
+ * https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem
+ */
+function base64ToBytes(base64: string) {
+  const binString = atob(base64)
+  return Uint8Array.from(binString, (m) => m.codePointAt(0)!)
+}
+
+function bytesToBase64(bytes: Uint8Array) {
+  const binString = Array.from(bytes, (byte) =>
+    String.fromCodePoint(byte)
+  ).join('')
+  return btoa(binString)
+}
```

Comment:
Wow, btoa atob really sucks for the reason linked...Nice catch. While we _don't_ support anything outside the ASCII character set, it's good in case we ever do! Kills me JS doesn't just support this and you had to write it + the tests haha.

Reply:
Right?? I think we do support it don't we? People have written KCL with kanji in the comments I think? Not to mention emoji 😭

Reply:
No we technically don't and we've have our KCL parser explode because of it :p